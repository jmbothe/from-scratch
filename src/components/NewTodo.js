import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';

import NewTodoField from './NewTodoField';

export class NewTodo extends Component {
  render() { 
    return (
    <section>
      <form>
        <Field name="new-todo-input" component={NewTodoField} />
      </form>
    </section>
    )
  }
}

export default NewTodo;