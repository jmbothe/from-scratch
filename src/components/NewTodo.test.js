import React from 'react';
import { shallow } from 'enzyme';

import { NewTodo } from './NewTodo';

describe('<NewTodo />', () => {
  const newTodo = shallow(<NewTodo />)

  it('renders correctly', () => {
    expect(newTodo).toMatchSnapshot();
  })

  it('renders a form element', () => {
    expect(newTodo.find('form').exists()).toBe(true);
  })

  it('renders a redux-form <Field />', () => {
    expect(newTodo.find('Field').exists()).toBe(true);
  })
})