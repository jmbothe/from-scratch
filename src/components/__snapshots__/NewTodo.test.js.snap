// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewTodo /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewTodo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <Field
          component={[Function]}
          name="new-todo-input"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Field
          component={[Function]}
          name="new-todo-input"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "name": "new-todo-input",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <Field
            component={[Function]}
            name="new-todo-input"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Field
            component={[Function]}
            name="new-todo-input"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "name": "new-todo-input",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
